### ISSUE TITLE
Add support for C# 10 implicit global usings 

### ISSUE THREAD

**Issue by JeremyCaney** [Ignia/Topics-Library#93]
C# 10 introduced support for [Implicit Global Usings](https://docs.microsoft.com/en-us/dotnet/core/project-sdk/overview#implicit-using-directives), which automatically add common namespaces based on the target SDK in the `csproj` file. This can reduce the number of `using` statements needed on each class.

### Enabling
Implicit global usings can be enabled in the `csproj` file by setting the following property:
```xml
<PropertyGroup>
  <ImplicitUsings>enable</ImplicitUsings>
</PropertyGroup>
```

### Limitations
While this is advertised as a C# 10 feature, it's actually dependent on the .NET 6 SDK, due to tooling integrated with `dotnet` (and, under the hood, `msbuild`). Since we have no intention of upgrading the OnTopic library to .NET 6 anytime soon (in order to maintain backward compatibility) this could be a problem. That said, so long as the latest version of the .NET SDK is used to compile the project, this may still be supported. TBD.

### Workaround
If this is _not_ supported without upgrading the project to .NET 6, we can work around it by instead taking advantage of C# 10's global using statements and _explicitly_ setting up the ones that come with the SDK—as well as any others we want to take advantage of. 

### Implicit Using Directives
With the `Microsoft.NET.Sdk`, we get the following implicit usings:

```cs
global using System;
global using System.Collections.Generic;
global using System.IO;
global using System.Linq;
global using System.Net.Http;
global using System.Threading;
global using System.Threading.Tasks;
```

With the `Microsoft.NET.Sdk.Web`, we additionally get:

```cs 
global using System.Net.Http.Json
global using Microsoft.AspNetCore.Builder
global using Microsoft.AspNetCore.Hosting
global using Microsoft.AspNetCore.Http
global using Microsoft.AspNetCore.Routing
global using Microsoft.Extensions.Configuration
global using Microsoft.Extensions.DependencyInjection
global using Microsoft.Extensions.Hosting
global using Microsoft.Extensions.Logging
```

**Comment by JeremyCaney** [comment]
***Update:*** So long as the .NET 6 SDK (`dotbuild`) is used to compile the project, implicit usings are supported, even if the project itself is not targeting .NET 6. This will necessitate ensuring the correct SDK is used as part of the CI/CD pipeline on Azure DevOps. Obviously, the project _does_ need to be using C# 10.

**Comment by JeremyCaney** [comment]
In addition to implementing C# 10 implicit usings on each project, I also ended up implementing C# 10 global using statements. For the latter, I identified the top 3-5 using statements in the project, and generally targeted those impacting more than half of the files. These were then added to the `AssemblyInfo` class, since that's already used for assembly-wide settings. 

For more information, see c9f6b179.
